internal/access/longpolling.go

```diff
@@ -0,0 +1,67 @@
+package access
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/infrahq/infra/internal"
+	"github.com/infrahq/infra/internal/logging"
+	"github.com/infrahq/infra/internal/server/data"
+)
+
+type updateIndexable interface {
+	UpdateIndex() int64
+	ItemCount() int
+}
+
+func blockingRequest[Result updateIndexable](rCtx RequestContext, listenOpts data.ListenForNotifyOptions, query func() (Result, error), lastUpdateIndex int64) (Result, error) {
+
+	listener, err := data.ListenForNotify(rCtx.Request.Context(), rCtx.DataDB, listenOpts)
+	if err != nil {
+		// nolint: gocritic
+		return *new(Result), fmt.Errorf("listen for notify: %w", err)
```

Comment:
Why return a new result like this?

Reply:
AFAIK there are not a lot of ways to create a new instance of a generics type.

Reply:
I think we might be able to return `nil` since it's an interface